### DISCUSSION TITLE
setInterval() giving unexpected results while updating the state of React components

### DISCUSSION THREAD

**Discussion by ToptenDev** [community/community#58889] [Programming Help] [answered]
### Body

I am trying to create a counter which decreases every second after clicking on it. The initial state of the clock is 25:00 which is rendered as {timerMinutes}:{timerSeconds} in the Timer component. I am decrementing the previous number of seconds by 1 to calculate the new state and based on this new state, the new minutes and seconds are calculated.

However, the clock goes back to 24:59 after 24:00 and keeps looping in that range forever.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by iamsuiux** [comment, marked as answer]
I think you're making it way harder on yourself by using two states that are that closely connected. Something like this seems to work:

```
const [timer, setTimer] = useState({ minutes: 25, seconds: 0});

const startStop = () => {
  setInterval(()=>{
    setTimer(({ minutes, seconds }) => {
      let time = minutes*60 + seconds;
      time-=1;

      return {
        minutes: Math.floor(time / 60),
        seconds: time % 60,
      };
    });
  }, 1000);
}
```
